include/nlohmann/json.hpp

```diff
@@ -3944,7 +3860,7 @@ class basic_json // NOLINT(cppcoreguidelines-special-member-functions,hicpp-spec
             const auto it = find(key);
             if (it != end())
             {
-                return it->template get<ValueType>();
+                return it->template get<typename std::decay<ValueType>::type>();
             }
 
             return default_value;
```

Comment:
I'm pretty sure, this should be  `return std::forward<ValueType>( default_value )` to avoid unnecessary copies, if `default_value` is a temporary. 

Thats something that even could/should be fixed in the main branch (where `default_value` is currently always a `const ValueType&`) but I don't want to create a separate PR that conflicts with this one  and would be made obsolete by it.

Reply:
Fixed.

Reply:
Apparently thats not the only place, where forwarding should be applied: ci_test_clang  fails with a warning about that in line 3942. Sorry I didn't spot that immediately.

I think that contrary to the suggestion by clang, you should again use std::forward and not std::move.

Also, I think the whole function should get the same `std::decay_t` treatment as the previous overload of `value`.

Reply:
Fixed.